src/util/dictionary.cpp

```diff
@@ -0,0 +1,358 @@
+//  SuperTux
+//  Copyright (C) 2016 Christian Hagemeier <christian@hagemeier.ch>
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#include "util/dictionary.hpp"
+#include <iostream>
+
+using std::vector;
+using std::cout;
+using namespace boost;
+using std::auto_ptr;
+int
+dictionary::hash(std::string key)
+{
+  return ((int) key[0])%10;
+}
+
+dictionary::dictionary(const std::string& filename):
+m_table(),
+m_filename(),
+m_writer()
+{
+  m_filename = filename;
+  // Load from file if it exists
+  this->load();
+}
+
+dictionary::dictionary(void):
+m_table(),
+m_filename(),
+m_writer()
+{
+}
+
+dictionaryTypes dictionary::str2type(std::string type)
```

Comment:
Is this a static method? If not, mark this const.

Also, parameter should be const std::string&